### DISCUSSION TITLE
Android app "Download new version" button is not clickable

### DISCUSSION THREAD

**Discussion by zodmy** [rustdesk/rustdesk#9407] [General]
### Bug Description

Title describes the issue 

### How to Reproduce

Have an older version of the app

### Expected Behavior

Button is clickable or changed Ukrainian translation

### Operating system(s) on local side and remote side

Android

### RustDesk Version(s) on local side and remote side

1.2.7

### Screenshots

![Screenshot_20240919_230756](https://github.com/user-attachments/assets/60d4aab8-f3a9-4b9a-8cfe-12a0095ac336)

### Additional Context

If this is not a bug, should the Ukrainian translation of the button text be changed to "New version available" or as current English "Download new version", as the current Ukrainian means "To download new version" (which is why it is expected to be clickable)?

**Comment by kramttocs** [comment]
What is the status of this?

Additional, the rustdesk.com text in the About page isn't a link which one would expect

**Reply by zodmy** [reply]
Still relevant

**Comment by zodmy** [comment]
I think the button is designed to be clickable
https://github.com/rustdesk/rustdesk/blob/f38d89aaeefd008689bc98b11c2559576636c669/flutter/lib/mobile/pages/connection_page.dart#L118-L138

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/pull/10165

**Comment by zycboss** [comment]
Not fixed yet, 1.3.6 is still not clickable

**Reply by zodmy** [reply]
It's clickable on 1.3.5

**Reply by zycboss** [reply]
> It's clickable on 1.3.5

Not for me

Samsung S24

https://github.com/user-attachments/assets/9a53ca75-e83c-4128-8ade-088ba416c6c4

**Reply by zodmy** [reply]
> > It's clickable on 1.3.5
> 
> Not for me
> 
> Samsung S24
> 
> https://github.com/user-attachments/assets/9a53ca75-e83c-4128-8ade-088ba416c6c4
> 
> 

Samsung A51

https://github.com/user-attachments/assets/413ad71c-8a24-4e80-88ad-1b0b84688ecf

**Reply by rustdesk** [reply]
@zycboss I see you know flutter. can you debug this since you are easy to reproduce this? You can create a new flutter project for this, since RustDesk is too heavy for you to test.

The code is quite simple.

https://github.com/rustdesk/rustdesk/blob/39a430f96f9600a2ef19a7191d3f323c7ba4b8b5/flutter/lib/mobile/pages/connection_page.dart#L118

It seems missing some settings.

https://pub.dev/packages/url_launcher#:~:text=cases%20starting%20on-,Android%2011%20(API%2030),-or%20higher.%20Checking

https://developer.android.com/training/package-visibility/use-cases?hl=zh-cn#:~:text=%E6%98%BE%E5%BC%8F%20intent%E3%80%82-,%E6%A3%80%E6%9F%A5%E6%B5%8F%E8%A7%88%E5%99%A8%E6%98%AF%E5%90%A6%E5%8F%AF%E7%94%A8,-%E5%9C%A8%E6%9F%90%E4%BA%9B%E6%83%85%E5%86%B5

**Reply by 21pages** [reply]
Could you try this test program to see if it can be clicked and navigate to the download page? (This program has a download button that is forced to display.)

https://github.com/21pages/rustdesk/releases/download/android-update/rustdesk-1.3.6-aarch64.apk

https://github.com/21pages/rustdesk/commit/a70da6079df80f5cf852fbd3f08352af0c7b46f1

**Reply by zycboss** [reply]
> Could you try this test program to see if it can be clicked and navigate to the download page? (This program has a download button that is forced to display.)
> 
> https://github.com/21pages/rustdesk/releases/download/android-update/rustdesk-1.3.6-aarch64.apk
> 
> [21pages@a70da60](https://github.com/21pages/rustdesk/commit/a70da6079df80f5cf852fbd3f08352af0c7b46f1)

Yes, this version works

**Reply by 21pages** [reply]
How about this one? https://github.com/21pages/rustdesk/releases/download/android-update2/rustdesk-1.3.6-aarch64.apk

Some code changes may be unnecessary, https://github.com/21pages/rustdesk/commit/55709a558a2a5b49672c26be6a5ccadad3fe0357

**Reply by zycboss** [reply]
> How about this one? https://github.com/21pages/rustdesk/releases/download/android-update2/rustdesk-1.3.6-aarch64.apk
> 
> Some code changes may be unnecessary, [21pages@55709a5](https://github.com/21pages/rustdesk/commit/55709a558a2a5b49672c26be6a5ccadad3fe0357)

This one works too

**Comment by zycboss** [comment]
> The fix has been merged, but I want to confirm whether canLaunchUrl works after changing manifest, could you try out https://github.com/21pages/rustdesk/releases/download/android-update3/rustdesk-1.3.6-aarch64.apk to see if it still works?

Thanks for your efforts, this one still works.
